rs/canister_sandbox/src/replica_controller/sandboxed_execution_controller.rs

```diff
@@ -1721,13 +1739,16 @@ fn evict_sandbox_processes(
         Backend::Empty => false,
     });
 
+    let priorities = state_reader.get_latest_scheduler_priorities();
```

Comment:
[This is a high-level comment, but I'm putting it here because GitLab can only make threads out of comments on code.]

Given thousands and thousands of persistently active canisters (e.g. thousands of canisters with heartbeats) this will clearly outperform a LRU eviction scheme. But I'm concerned that given a less regular pattern (e.g. lots of canisters receiving messages at random times, but with something like a [Pareto distribution](https://en.wikipedia.org/wiki/Pareto_distribution), which I would argue is a rather common case, including with timers) this might behave significantly worse than LRU (since the scheduler essentially goes round-robin over canisters, making any sandbox as likely to be evicted as any other; whereas LRU is as close to optimal as possible in that particular situation).

Have you considered the trade-off? And/or thought of alternative eviction schemes that may not exhibit pathologically bad behavior in common situations?

Reply:
That is a good observation. And yes, we did consider the tradeoff, and the execution team believes it is worth it.

The case we were testing is a very common case, where we have a huge number of heartbeats, and it was observed on the subnets that were in focus due to intensive load. However, I agree that the case you present may be common too.

Furthermore, after releasing it on the mainnet we plan to observe the behavior. 

Additionally, I had an idea about keeping all candidates executed in the previous round and after that candidates based on the scheduler's priorities. (or have a constant that represents the ratio between the candidates kept based on `last_executed` and `scheduler_priority`). 

But we plan to only add priority-based eviction to new release, so we can have a better understanding of how change behaves when it is isolated. Based on that we will decide on the following steps.